pkg/object/httpserver/spec.go

```diff
@@ -136,8 +136,11 @@ func (spec *Spec) tlsConfig() (*tls.Config, error) {
 		Certificates: certificates,
 		NextProtos:   []string{"acme-tls/1"},
 	}
+
+	acm := autocertmanager.GetGlobalAutoCertManager()
+
 	tlsConf.GetCertificate = func(chi *tls.ClientHelloInfo) (*tls.Certificate, error) {
-		return autocertmanager.GetCertificate(chi, !spec.AutoCert /* tokenOnly */)
+		return acm.GetCertificate(chi, !spec.AutoCert /* tokenOnly */)
```

Comment:
could `acm` be nil here?

Reply:
In real running mode, it won't be, but I add a line to do defensive programming.

Reply:
We chose to make AutoCertManager to BusinessController because its properties do not seem like a system controller.